aten/src/ATen/Formatting.cpp

```diff
@@ -45,10 +45,6 @@ std::ostream& operator<<(std::ostream & out, Backend b) {
   return out << toString(b);
 }
 
-std::ostream& operator<<(std::ostream & out, ScalarType t) {
```

Comment:
why this change?

Reply:
The short answer is, good C++ practice is to put `operator<<` for user defined types in the same namespace as the type in question.

The long answer is, C++ has a feature called argument dependent lookup, whereby when it is looking for an implementation of `operator<<` to satisfy something like `std::cout << scalar_type`, it collects the namespaces of all of its arguments and looks for `operator<<` in those namespaces. This is why this program compiles: https://godbolt.org/g/a4ZQyK even though, ordinarily, `f::operator<<` would not be considered "in-scope" for a normal function call.

Now, in many cases, whether or not you put the `operator<<` definition in the global namespace or in your user namespace doesn't matter, because scope resolution will find the right one anyway. But if you have some `operator<<` definitions in a namespace, and you perform a `<<` in that namespace, the compiler will see that the namespaced `operator<<` is the relevant one, and will ignore your overloads in the global namespace. Example: https://godbolt.org/g/eZjXEs

You can "work around" this problem by explicitly telling the compiler to use the global `operator<<` with a using declaration https://godbolt.org/g/PjvoSR but there is a simpler, more excellent way to fix the problem: DON'T define `operator<<` in the global namespace; instead, rely on ADL to figure out what the correct namespace of `operator<<` should be.

(Also CC @cpuhrsch, this is why you should do it this way!)